src/lib/modules/tests/test.js

```diff
@@ -268,7 +269,9 @@ class Test {
               self.contracts[contract.className] = {};
             }
 
-            const newContract = Test.getWeb3Contract(contract, web3);
+            const testContractFactoryPlugin = self.plugins.getPluginsFor('testContractFactory').slice(-1)[0];
```

Comment:
As mentioned above, we're always taking the last one that has been registered as there can be only one. Hacky feelings coming up here when reading this code. =/

Reply:
If there's more than one, could we chain them together, e.g. the return value of the first becomes input to the second, and so on (or reverse order)?

Reply:
I actually believe that could break many scenarios.

This factory function allows you to do whatever you want. So if we pipe the return value in to other functions, they can never know what exactly they'll be getting. 

I'm also not sure whether there would be a reasonable scenario to have multiple plugins that would alter this behaviour. I'm pretty close to saying that this should be a singleton-last-one-wins setting..